Assets/SEE/Game/SceneManipulation/SpatialMetrics.cs

```diff
@@ -0,0 +1,253 @@
+                }
+                throw new IndexOutOfRangeException($"Given direction is not possible in {nameof(Bounds2D)}: {dir}");
+            }
+
+            set {
+                if (dir == Vector3.left)
+                {
+                    Left = value;
+                }
+                else if (dir == Vector3.right)
+                {
+                    Right = value;
+                }
+                else if (dir == Vector3.back)
+                {
+                    Back = value;
+                }
+                else if (dir == Vector3.forward)
+                {
+                    Front = value;
+                }
+                throw new IndexOutOfRangeException($"Given direction is not possible in {nameof(Bounds2D)}: {dir}");
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the element at the specified direction.
+        /// </summary>
+        /// <param name="dir">The direction of the element to get or set.</param>
+        /// <returns>The element at the specified direction.</returns>
+        /// <exception cref="IndexOutOfRangeException">
+        /// Thrown when the direction is <see cref="Direction2D.None"/>.
+        /// </exception>
+        public float this[Direction2D dir]
+        {
+            get {
+                switch (dir)
+                {
+                    case Direction2D.Left:
+                        return Left;
+                    case Direction2D.Right:
+                        return Right;
+                    case Direction2D.Back:
+                        return Back;
+                    case Direction2D.Front:
+                        return Front;
+                    default:
+                        throw new IndexOutOfRangeException($"Given direction is not possible in {nameof(Bounds2D)}: {dir}");
+                }
+            }
+
+            set {
+                switch (dir)
+                {
+                    case Direction2D.Left:
+                        Left = value;
+                        break;
+                    case Direction2D.Right:
+                        Right = value;
+                        break;
+                    case Direction2D.Back:
+                        Back = value;
+                        break;
+                    case Direction2D.Front:
+                        Front = value;
+                        break;
+                    default:
+                        throw new IndexOutOfRangeException($"Given direction is not possible in {nameof(Bounds2D)}: {dir}");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Implicit conversion to string.
+        /// <summary>
+        public static implicit operator string(Bounds2D bounds)
+        {
+            return bounds.ToString();
+        }
+
+        /// <summary>
+        /// Returns a printable string with the struct's values.
+        /// <summary>
+        public override readonly string ToString()
```

Comment:
Order should be `readonly override`:
```suggestion
        public readonly override string ToString()
```